backends/rapidpro/msg.go

```diff
@@ -181,17 +183,37 @@ func readMsgFromDB(b *backend, id courier.MsgID) (*DBMsg, error) {
 // Media download and classification
 //-----------------------------------------------------------------------------
 
-func downloadMediaToS3(b *backend, orgID OrgID, msgUUID courier.MsgUUID, mediaURL string) (string, error) {
+func downloadMediaToS3(ctx context.Context, b *backend, channel courier.Channel, orgID OrgID, msgUUID courier.MsgUUID, mediaURL string) (string, error) {
+
 	parsedURL, err := url.Parse(mediaURL)
 	if err != nil {
 		return "", err
 	}
 
-	// first fetch our media
-	req, err := http.NewRequest("GET", mediaURL, nil)
-	if err != nil {
-		return "", err
+	var req *http.Request
+	handler := courier.GetHandler(channel.ChannelType())
+	if handler != nil {
+		mediaDownloadRequestBuilder, isMediaDownloadRequestBuilder := handler.(courier.MediaDownloadRequestBuilder)
```

Comment:
I like this approach, how about some shorter variable names? Go's style says use the shortest possible names.

So just using `builder` and `isBuilder` here would be better for example.